crates/matrix-sdk/src/event_cache/store.rs

```diff
@@ -230,6 +50,11 @@ impl RoomEvents {
         Self { chunks: LinkedChunk::new() }
     }
 
+    /// Clear all events.
+    pub fn reset(&mut self) {
+        self.chunks = LinkedChunk::new();
```

Comment:
Hmm I don't think it's worth the effort. Ultimately, the `reset` must not be used. We use it now because we don't the reconciliation algorithm yet, but it's not the final desired situation.

And recycling `Chunk` is going to be quite complex honestly… (since it's a pointer-based structure, the risk of messing things up is not zero 😛). I understand the reason, but let's not do some premature optimisation here.